### ISSUE TITLE
Resnet18,cifar10.

### ISSUE THREAD

**Issue by oneheartyx** [TsingZ0/PFLlib#220]
![Uploading issue.png…]()

感谢你们开源了如此有用的代码！但是我在Resnet上训练Cifar10的时候，使用了之前issue中提到的参数，round number=3时acc只有0.2276，与图片中的0.4233有较大差距，请问这是什么原因造成的

-------------Round number: 0-------------

Evaluate global model
Averaged Train Loss: 2.3069
Averaged Test Accurancy: 0.0960
Averaged Test AUC: 0.5071
Std Test Accurancy: 0.0108
Std Test AUC: 0.0083
------------------------- time cost ------------------------- 43.83143663406372

-------------Round number: 1-------------

Evaluate global model
Averaged Train Loss: 2.3517
Averaged Test Accurancy: 0.1194
Averaged Test AUC: 0.5492
Std Test Accurancy: 0.0202
Std Test AUC: 0.0137
------------------------- time cost ------------------------- 42.82741069793701

-------------Round number: 2-------------

Evaluate global model
Averaged Train Loss: 2.2278
Averaged Test Accurancy: 0.1713
Averaged Test AUC: 0.6381
Std Test Accurancy: 0.0268
Std Test AUC: 0.0241
------------------------- time cost ------------------------- 42.945683002471924

-------------Round number: 3-------------

Evaluate global model
Averaged Train Loss: 2.0694
Averaged Test Accurancy: 0.2276
Averaged Test AUC: 0.7141
Std Test Accurancy: 0.0358
Std Test AUC: 0.0308
------------------------- time cost ------------------------- 43.00451135635376

-------------Round number: 4-------------

Evaluate global model
Averaged Train Loss: 1.8288
Averaged Test Accurancy: 0.3223
Averaged Test AUC: 0.7794
Std Test Accurancy: 0.0302
Std Test AUC: 0.0184
------------------------- time cost ------------------------- 44.888002157211304

**Comment by TsingZ0** [comment]
感谢关注！之后代码有所调整，建议重新调参，谢谢！

**Comment by oneheartyx** [comment]
一心
***@***.***

很高兴能够收到您的回信！
关于IID情况下FedAvg的测试精度问题，我调试了很多遍代码，他的精度一致上不去，一开始我以为是过拟合了，但我用不用数据增强，精度始终上不到80以上。而且使用大或能麻烦您解答一下吗？
具体情况如下
-------------Round number: 996-------------
24-11-08 17:46:29.912 :&nbsp;
Evaluate global model
24-11-08 17:46:42.619 : Averaged Train Loss: 0.0053
24-11-08 17:46:42.620 : Averaged Test Accurancy: 0.6768
24-11-08 17:46:42.620 : Averaged Test AUC: 0.9323
24-11-08 17:46:42.620 : Std Test Accurancy: 0.0255
24-11-08 17:46:42.620 : Std Test AUC: 0.0080
24-11-08 17:47:15.336 : ------------------------- time cost ------------------------- 45.44329309463501
24-11-08 17:47:15.355 :&nbsp;
-------------Round number: 997-------------
24-11-08 17:47:15.355 :&nbsp;
Evaluate global model
24-11-08 17:47:28.191 : Averaged Train Loss: 0.0071
24-11-08 17:47:28.191 : Averaged Test Accurancy: 0.6786
24-11-08 17:47:28.191 : Averaged Test AUC: 0.9322
24-11-08 17:47:28.192 : Std Test Accurancy: 0.0257
24-11-08 17:47:28.192 : Std Test AUC: 0.0082
24-11-08 17:48:00.809 : ------------------------- time cost ------------------------- 45.47278571128845
24-11-08 17:48:00.828 :&nbsp;
-------------Round number: 998-------------
24-11-08 17:48:00.828 :&nbsp;
Evaluate global model
24-11-08 17:48:13.608 : Averaged Train Loss: 0.0058
24-11-08 17:48:13.608 : Averaged Test Accurancy: 0.6757
24-11-08 17:48:13.608 : Averaged Test AUC: 0.9320
24-11-08 17:48:13.608 : Std Test Accurancy: 0.0248
24-11-08 17:48:13.608 : Std Test AUC: 0.0080
24-11-08 17:48:43.623 : ------------------------- time cost ------------------------- 42.813679695129395
24-11-08 17:48:43.642 :&nbsp;
-------------Round number: 999-------------
24-11-08 17:48:43.642 :&nbsp;
Evaluate global model
24-11-08 17:48:55.841 : Averaged Train Loss: 0.0070
24-11-08 17:48:55.841 : Averaged Test Accurancy: 0.6768
24-11-08 17:48:55.841 : Averaged Test AUC: 0.9319
24-11-08 17:48:55.841 : Std Test Accurancy: 0.0281
24-11-08 17:48:55.841 : Std Test AUC: 0.0082
24-11-08 17:49:25.051 : ------------------------- time cost ------------------------- 41.428138256073
24-11-08 17:49:25.071 :&nbsp;
-------------Round number: 1000-------------
24-11-08 17:49:25.071 :&nbsp;
Evaluate global model
24-11-08 17:49:37.920 : Averaged Train Loss: 0.0076
24-11-08 17:49:37.921 : Averaged Test Accurancy: 0.6765
24-11-08 17:49:37.921 : Averaged Test AUC: 0.9320
24-11-08 17:49:37.921 : Std Test Accurancy: 0.0259
24-11-08 17:49:37.921 : Std Test AUC: 0.0079
24-11-08 17:50:07.709 : ------------------------- time cost ------------------------- 42.65790629386902
24-11-08 17:50:07.709 :&nbsp;
Best accuracy.
24-11-08 17:50:07.709 : 0.681
24-11-08 17:50:07.709 :&nbsp;
Average time cost per round.
24-11-08 17:50:07.709 : 44.28523062491417
24-11-08 17:50:07.754 :&nbsp;
Average time cost: 44334.86s.
24-11-08 17:50:07.755 : All done!

我的参数是
24-11-08 05:31:12.893 : goal = test
24-11-08 05:31:12.893 : device = cuda
24-11-08 05:31:12.893 : device_id = 0
24-11-08 05:31:12.893 : dataset = Cifar10
24-11-08 05:31:12.893 : num_classes = 10
24-11-08 05:31:12.893 : model = ResNet18
24-11-08 05:31:12.893 : batch_size = 16
24-11-08 05:31:12.893 : local_learning_rate = 0.1
24-11-08 05:31:12.893 : learning_rate_decay = True
24-11-08 05:31:12.893 : learning_rate_decay_gamma = 0.99
24-11-08 05:31:12.893 : global_rounds = 1000
24-11-08 05:31:12.893 : local_epochs = 1
24-11-08 05:31:12.893 : algorithm = FedAvg
24-11-08 05:31:12.893 : join_ratio = 1.0
24-11-08 05:31:12.893 : random_join_ratio = False
24-11-08 05:31:12.893 : num_clients = 20
数据增强：
normalize = transforms.Normalize(mean=[0.507, 0.487, 0.441], std=[0.267, 0.256, 0.276])
transform_train = transforms.Compose([transforms.RandomCrop(32, padding=4),
                                      transforms.RandomHorizontalFlip(),
                                      transforms.ColorJitter(brightness=0.25, contrast=0.8),
                                      transforms.ToTensor(),
                                      normalize,
                                      ])
transform_test = transforms.Compose([transforms.CenterCrop(32),
                                     transforms.ToTensor(),
                                     normalize,
                                     ]) 谢谢您的回答！  祝好！ 一心



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "TsingZ0/PFLlib"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年11月10日(星期天) 晚上8:05
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [TsingZ0/PFLlib] Resnet18,cifar10. (Issue #220)





 
感谢关注！之后代码有所调整，建议重新调参，谢谢！
